security updates needed in build container

# Bug

## Description

`test_ensure_no_updates_avail` is failing when building deb packages due to security updates needed in the build container

## Steps to Reproduce

`make build-debs`

##  Expected Behavior

All tests pass, all is well

## Actual Behavior

```
    _________ test_ensure_no_updates_avail[docker://trusty-sd-sec-update] __________
    [gw1] darwin -- Python 2.7.14 /Users/redshiftzero/.virtualenvs/securedrop/bin/python2.7
    
    host = <testinfra.host.Host object at 0x1098a7f10>
    
        def test_ensure_no_updates_avail(host):
            """
                Test to make sure that there are no security-updates in the
                base builder container.
            """
        
            # Filter out all the security repos to their own file
            # without this change all the package updates appeared as if they were
            # coming from normal ubuntu update channel (since they get posted to both)
            host.run('egrep "^deb.*security" /etc/apt/sources.list > /tmp/sec.list')
        
            dist_upgrade_simulate = host.run('apt-get -s dist-upgrade '
                                             '-oDir::Etc::Sourcelist=/tmp/sec.list '
                                             '|grep "^Inst" |grep -i security')
        
            # If the grep was successful that means security package updates found
            # otherwise we get a non-zero exit code so no updates needed.
    >       assert dist_upgrade_simulate.rc != 0
    E       AssertionError: assert 0 != 0
    E        +  where 0 = CommandResult(command='apt-get -s dist-upgrade -oDir::Etc::Sourcelist=/tmp/sec...0.7] (1:1.9.1-1ubuntu0.8 Ubuntu:14.04/trusty-security [amd64])\n', stderr=None).rc
    
    tests/test_security_updates.py:21: AssertionError
    ============== 1 failed, 175 passed, 1 xpassed in 105.93 seconds ===============
```

--- a/docs/development/dockerbuildmaint.rst
+++ b/docs/development/dockerbuildmaint.rst
@@ -1,0 +1,69 @@
+Docker Build Maintenance
+========================
+
+Get your Quay account squared away
+-----------------------------------
+The container that performs builds of debian packages is version controlled in
+a docker repository at **quay.io/freedomofpress/sd-docker-builder**.
+There are tight restrictions over who can make edits here. If you have permissions
+to do so, you'll need to make sure your local docker client has credentials to push.
+
+* First login into your quay.io account via the web-portal at https://quay.io/
+* Drill into your **Account settings** via the upper right drop-down (where your username is)
+* Click **Generate Encrypted Password**
+* From a command-line prompt type **docker login quay.io** with your username and credentials
+  obtained from the previous step.
+* Proceed with update instructions
+
+
+Performing container updates
+----------------------------
+If one of the dependencies requires security updates, the build may fail at
+**test_ensure_no_updates_avail** . If you have access rights to push to quay.io,
+here is the process to build and push a new container:
+
+.. code:: sh
+
+    cd molecule/builder/
+    # Build a new container
+    make build-container
+
+You should get a date tag as a result of the last command above, an example
+would be:
+
+.. code:: sh
+
+    Successfully tagged quay.io/freedomofpress/sd-docker-builder:2018_06_06
+
+After you have your new container built, to locally test it, update
+**molecule/builder/create.yml** with the following sed one-liner:
+
+.. code:: sh
+
+    # Replace 2018_06_06 with the tag received above
+    sed -i 's/@sha256:{{image_hash}}/:2018_06_06/g' create.yml
+    cd ../../
+    make build-deps
+
+
+Assuming no errors here, now push the image to **quay.io** and update the molecule
+scenario accordingly.
+
+.. code:: sh
+
+    # back we go to the molecule directory
+    cd molecule/builder
+    # clear out any potential edits you made while testing
+    git checkout .
+    # push the container you previously built
+    make push-container
+    # You'll get a digest here copy and replace that in the create playbook
+    # for example i received the following mesage
+    # 2018_06_06: digest: sha256:15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c size: 7907
+    sed -i "s/docker-builder\:[0-9]*.*/docker-builder\:2018_06_06/g" molecule/builder/create.yml
+    sed -i 's/image_hash:.*/image_hash: 15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c/g' create.yml
+
+
+Make sure your changes get commited and merged in. Others will have to re-base on that to take advantage of your changes.
+
+

--- a/docs/development/dockerbuildmaint.rst
+++ b/docs/development/dockerbuildmaint.rst
@@ -3,7 +3,7 @@ Docker Build Maintenance
 
 Get your Quay account squared away
 -----------------------------------
-The container that performs builds of debian packages is version controlled in
+The container that performs builds of Debian packages is version controlled in
 a docker repository at **quay.io/freedomofpress/sd-docker-builder**.
 There are tight restrictions over who can make edits here. If you have permissions
 to do so, you'll need to make sure your local docker client has credentials to push.
@@ -43,7 +43,7 @@ After you have your new container built, to locally test it, update
     # Replace 2018_06_06 with the tag received above
     sed -i 's/@sha256:{{image_hash}}/:2018_06_06/g' create.yml
     cd ../../
-    make build-deps
+    make build-debs
 
 
 Assuming no errors here, now push the image to **quay.io** and update the molecule
@@ -61,9 +61,7 @@ scenario accordingly.
     # for example i received the following mesage
     # 2018_06_06: digest: sha256:15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c size: 7907
     sed -i "s/docker-builder\:[0-9]*.*/docker-builder\:2018_06_06/g" molecule/builder/create.yml
-    sed -i 's/image_hash:.*/image_hash: 15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c/g' create.yml
+    sed -i 's/image_hash:.*/image_hash: 15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c/g' molecule/builder/create.yml
 
 
-Make sure your changes get commited and merged in. Others will have to re-base on that to take advantage of your changes.
-
-
+Make sure your changes get committed and merged in. Others will have to re-base on that to take advantage of your changes.

--- a/docs/index.rst
+++ b/docs/index.rst
@@ -118,6 +118,7 @@ anonymous sources.
    development/apparmor_profiles
    development/portable_demo
    development/release_management
+   development/dockerbuildmaint
 
 .. toctree::
   :caption: Threat Model

--- a/molecule/builder/Makefile
+++ b/molecule/builder/Makefile
@@ -1,4 +1,4 @@
-DATE_STR := $(shell date +"%Y_%m_%e")
+DATE_STR := $(shell date +"%Y_%m_%d")
 
 .PHONY: build-container
 build-container: ## Build container utilized for debian pkg building

--- a/molecule/builder/create.yml
+++ b/molecule/builder/create.yml
@@ -9,7 +9,7 @@
     molecule_scenario_directory: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}"
     molecule_yml: "{{ lookup('file', molecule_file) | from_yaml }}"
     # sha256 digest of quay.io/freedomofpress/sd-docker-builder:2018_06_06
-    image_hash: 15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c
+    image_hash: 7a5f4504f45738a004ca47f6a01e51601e5f9ab476899778b065fac8f91cdd1b
   tasks:
     - name: Create builders
       docker_container:

--- a/molecule/builder/create.yml
+++ b/molecule/builder/create.yml
@@ -8,8 +8,8 @@
     molecule_ephemeral_directory: "{{ lookup('env', 'MOLECULE_EPHEMERAL_DIRECTORY') }}"
     molecule_scenario_directory: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}"
     molecule_yml: "{{ lookup('file', molecule_file) | from_yaml }}"
-    # sha256 digest of quay.io/freedomofpress/sd-docker-builder:2018_05_23
-    image_hash: 14be07c3992fd14b7a0cc0137aabeaa97aada91a755c420b1b54a899b9d07567
+    # sha256 digest of quay.io/freedomofpress/sd-docker-builder:2018_06_06
+    image_hash: 15f43e8d86a164509bccbe9b1c9fb5e2b3e6edd87457a9b67fef47574ec8a89c
   tasks:
     - name: Create builders
       docker_container:

